client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/iteration/RemoteInnerPublisherHandler.java

```diff
@@ -112,7 +113,7 @@ protected long handleNextResponse(IterationNextResponse<K, E> nextResponse, Map.
 
    @Override
    protected void handleThrowableInResponse(Throwable t, Map.Entry<SocketAddress, IntSet> target) {
-      if (t instanceof TransportException || t instanceof RemoteIllegalLifecycleStateException) {
+      if (t instanceof TransportException || t instanceof RemoteIllegalLifecycleStateException || t instanceof ConnectException) {
```

Comment:
Wondering if it needs all of: https://github.com/infinispan/infinispan/blob/7534ad367e52ecc7a971f8b37f60b69367d89a50/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/RetryOnFailureOperation.java#L160-L163

Reply:
Hrmm, only ConnectException would be better if it is suitable.